lib/streamlit/caching/cache_utils.py

```diff
@@ -85,14 +130,51 @@ def cache_type(self) -> CacheType:
         raise NotImplementedError
 
     @property
-    def call_stack(self) -> "CachedFunctionCallStack":
+    def warning_call_stack(self) -> "CacheWarningCallStack":
+        raise NotImplementedError
+
+    @property
+    def message_call_stack(self) -> "CacheMessagesCallStack":
         raise NotImplementedError
 
     def get_function_cache(self, function_key: str) -> Cache:
         """Get or create the function cache for the given key."""
         raise NotImplementedError
 
 
+def replay_result_messages(result: CachedResult) -> None:
```

Comment:
Raise an explanatory error instead of letting the KeyNotFound bubble up

Reply:
When does this get called? it gets called when the key to a widget is not found?

Reply:
It gets called when a function's result was found in the cache